spec/enumerable_extension_spec.rb

```diff
@@ -2,132 +2,107 @@
 
 describe Enumerable do
 
-  before do
-    @test_array = (1..4).to_a
-  end
+  subject { (1..4).to_a }
 
   describe "#my_each" do
 
     it "doesn't modify original array" do
-      @test_array.my_each { |x| x + 12 }
+      subject.my_each { |x| x + 12 }
 
-      expect(@test_array).to eq([1,2,3,4])
+      expect(subject).to eq([1,2,3,4])
     end
 
     it "performs specified operation on all elements" do
-      expected = [2,3,4,5]
-      actual = []
-      @test_array.my_each { |x| actual << x + 1 }
+      result = []
+      subject.my_each { |x| result << x + 1 }
 
-      expect(actual).to eq(expected)
+      expect(result).to eq([2,3,4,5])
     end
 
   end
 
   describe "#my_each_with_index" do
 
     it "doesn't modify original array" do
```

Comment:
I think my_each_with_index should also epate this spec from my_each https://github.com/talihomz/Enumaruby/pull/2/files#diff-95d934febaaa9135d9c567171be87643R15

As `my_each_with_index` does exactly the same thing as `my_each` and in addition handles indices.